components/ikas/sources/common/base.mjs

```diff
@@ -0,0 +1,55 @@
+import ikas from "../../ikas.app.mjs";
+
+export default {
+  props: {
+    ikas,
+    http: {
+      type: "$.interface.http",
+      customResponse: false,
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async activate() {
+      const { data } = await this.ikas.makeRequest({
+        data: {
+          "query": `mutation {
+          saveWebhook(
+            input: {
+              scopes: "${this.getScope()}"
+              endpoint: "${this.http.endpoint}"
+            }
+          ) {
+            createdAt
+            deleted
+            endpoint
+            id
+            scope
+            updatedAt
+          }
+        }`,
+        },
+      });
+      this.db.set("webhookId", data.saveWebhook[0].id);
+    },
+    async deactivate() {
+      const webhookId = this.db.get("webhookId");
+      if (webhookId) {
+        await this.ikas.makeRequest({
+          data: {
+            "query": `mutation {
+            deleteWebhook(scopes: ["${this.getScope()}"])
+          }`,
+          },
+        });
+      }
+    },
+  },
+  async run({ body }) {
+    this.$emit(body, {
```

Comment:
If you parse `body.json` here, you could save that step in `getSummary()` in the sources, and it might make the emitted event more user-friendly.
```suggestion
    body = {
       ...body,
       data: JSON.parse(body.data),
    };
    this.$emit(body, {
```